dataikuapi/dss/wiki.py

```diff
@@ -0,0 +1,250 @@
+        self.client = client
+        self.project_key = project_key
+
+    def get_settings(self):
+        """
+        Get wiki settings
+
+        :returns: an handle to manage the wiki settings (taxonomy, home article)
+        :rtype: DSSWikiSettings
+        """
+        return DSSWikiSettings(self.client, self.project_key, self.client._perform_json("GET", "/projects/%s/wiki/" % (self.project_key)))
+
+    def get_article(self, article_id):
+        """
+        Get a wiki article
+
+        :param str article_id: the article ID
+        :returns: an handle to manage the Article
+        :rtype: DSSWikiArticle
+        """
+        return DSSWikiArticle(self.client, self.project_key, article_id)
+
+    def __flatten_taxonomy__(self, taxonomy):
+        """
+        Private recursive method to get the flatten list of article IDs from the taxonomy
+
+        :param list taxonomy:
+        :returns: list of articles
+        :rtype: list
+        """
+        article_list = []
+        for article in taxonomy:
+            article_list.append(self.get_article(article['id']))
+            article_list += self.__flatten_taxonomy__(article['children'])
+        return article_list
+
+    def list_articles(self):
+        """
+        Get a list of all the articles
+
+        :returns: list of articles
+        :rtype: list
+        """
+        return self.__flatten_taxonomy__(self.get_settings().get_taxonomy())
+
+    def create_article(self, article_id, parent_id=None, content=None):
+        """
+        Create a wiki article
+
+        :param str article_id: the article ID
+        :param str parent_id: the parent article ID (or None if the article has to be at root level)
+        :param str content: the article content
+        :returns: the created article
+        :rtype: DSSWikiArticle
+        """
+        body = {
+            "projectKey": self.project_key,
+            "id": article_id,
+            "parent": parent_id
+        }
+        self.client._perform_json("POST", "/projects/%s/wiki/" % (self.project_key), body=body)
+        article = DSSWikiArticle(self.client, self.project_key, article_id)
+
+        # set article content if given
+        if content is not None:
+            article_data = article.get_data()
+            article_data.set_body(content)
+            article_data.save()
+
+        return article
+
+class DSSWikiSettings(object):
+    """
+    Global settings for the wiki, including taxonomy. Call save() to save
+    """
+    def __init__(self, client, project_key, settings):
+        """
+        :param DSSClient client: an api client to connect to the DSS backend
+        :param str project_key: identifier of the project to access
+        :param dict settings: current wiki settings (containing taxonomy and home article)
+        """
+        self.client = client
+        self.project_key = project_key
+        self.settings = settings
+
+    def get_taxonomy(self):
+        """
+        Get the taxonomy
```

Comment:
This would require a bit more documentation about what kind of things are in the taxonomy. It seems that the helper "move article in taxonomy" is missing :)

Reply:
Also need to explain that you get a direct reference, not a copy, so modifications to the returned object will be reflected when saving